src/pages/index.js

```diff
@@ -4,51 +4,95 @@ import PropTypes from 'prop-types'
 import { graphql } from 'gatsby'
 
 export const query = graphql`
-  {
-    homepage: allCosmicjsHomes {
+  query HomePageQuery($locale: String) {
+    homePage: allPrismicHomepage(filter: { lang: { eq: $locale } }) {
```

Comment:
use prismicHomepage query instead of "all" prismicHomepage. This does nothing for this particular but a large project, it will save a bit of computing efforts